### PR TITLE
[Workspace] add use case cards when workspace is disabled, refine the style of recent_work in the home page.

### PR DESCRIPTION
**Pull request by Qxisylolo**
### Description
This pr can be divided as 2 parts:
1. add use case cards when workspace is disabled
2. refine the style of recent_work in the home page, fix the line break issue that occurs when the screen becomes excessively large.

## Screenshot
1. add use case cards when workspace is disabled
Screenshot 2024-09-03 at 13 05 23

2. fix the line break issue that occurs when the screen becomes excessively large
before:
Screenshot 2024-09-03 at 13 08 39

now:
Screenshot 2024-09-03 at 13 07 15

## Changelog
- feat: add use case cards when workspace is disabled, refine the style of recent_work in the home page.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/home/public/application/home_render.tsx [hunk 1]
```diff
@@ -48,27 +52,52 @@ export const setupHome = (contentManagement: ContentManagementPluginSetup) => {
...
+        id: SECTIONS.SERVICE_CARDS,
+        order: 3000,
+        kind: 'dashboard',
       },
     ],
   });
 };
 
 export const initHome = (contentManagement: ContentManagementPluginStart, core: CoreStart) => {
+  // registerUseCaseCard(contentManagement,work);
```
Comment (raintygao) [inline src/plugins/home/public/application/home_render.tsx:64]: Could we remove this line?

File: src/plugins/home/public/application/components/use_case_card.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      }),
+      description: i18n.translate('workspace.usecase.search.description', {
+        defaultMessage: 'Discover and query your data with ease',
+      }),
+      icon: 'wsSearch',
+    },
+  },
+};
+
+export const registerUseCaseCard = (
```
Comment (raintygao) [inline src/plugins/home/public/application/components/use_case_card.tsx:57]: Do we need to declare this file as `tsx` file? Seems that this file doesn't export any components or elements.

File: src/plugins/home/public/application/components/use_case_card.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+import { ContentManagementPluginStart } from '../../../../content_management/public';
+
+interface Config {
+  title: string;
+  description: string;
+  icon: string;
+}
+
+export const WORKSPACE_USE_CASES = {
```
Comment (raintygao) [inline src/plugins/home/public/application/components/use_case_card.tsx:18]: I think we may not need to maintain this config twice as it is similar to the `WORKSPACE_USE_CASES` in src/plugins/workspace/common/constants.ts. But I may also be doubtful about whether we should set home plugin depending on workspace.

**Comment by ruanyl** [comment]
Is this PR ready to review? Also could you please resolve the conflicts?

File: src/plugins/saved_objects_management/public/management_section/recent_work.tsx [hunk 4]
```diff
@@ -258,56 +258,65 @@ export const RecentWork = (props: { core: CoreStart; workspaceEnabled?: boolean
                   href={recentNavLink.href}
                   footer={
                     <>
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/management_section/recent_work.tsx:260]:
```suggestion
```
Nit: useless fragment

File: src/plugins/saved_objects_management/public/management_section/recent_work.tsx [hunk 5]
```diff
@@ -258,56 +258,65 @@ export const RecentWork = (props: { core: CoreStart; workspaceEnabled?: boolean
...
+                                  {i18n.translate(
+                                    'savedObjectsManagement.recentWorkSection.workspace',
+                                    {
+                                      defaultMessage: 'Workspace',
+                                    }
+                                  )}
+                                  :
+                                </EuiText>
+                              </EuiFlexItem>
+                              <EuiFlexItem grow={1} style={{ textAlign: 'right' }}>
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/management_section/recent_work.tsx:311]:
```suggestion
                              <EuiFlexItem grow={1} className="eui-textRight">
```
Nit:

File: src/plugins/saved_objects_management/public/management_section/recent_work.tsx [hunk 6]
```diff
@@ -258,56 +258,65 @@ export const RecentWork = (props: { core: CoreStart; workspaceEnabled?: boolean
...
+                                  : i18n.translate(
+                                      'savedObjectsManagement.recentWorkSection.updatedAt',
+                                      {
+                                        defaultMessage: 'Updated',
+                                      }
+                                    )}
+                                :{' '}
                               </EuiText>
                             </EuiFlexItem>
                             <EuiFlexItem grow={1} style={{ textAlign: 'right' }}>
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/management_section/recent_work.tsx:286]:
```suggestion
                            <EuiFlexItem grow={1}  className="eui-textRight">
```

File: src/plugins/home/public/application/components/use_case_card.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  }: {
+    target: string;
+    order: number;
+    id: string;
+    title: string;
+    description: string;
+    icon: string;
+  }
+) => {
+  contentManagement.registerContentProvider({
```
Comment (SuZhou-Joe) [inline src/plugins/home/public/application/components/use_case_card.ts:28]: Just to confirm that it's just a plain card without link?

File: src/plugins/saved_objects_management/public/management_section/recent_work.tsx (hunk 6 above)
Comment (Qxisylolo) [reply src/plugins/saved_objects_management/public/management_section/recent_work.tsx]: Thank you for the comments. I 'm sorry I have a later commit that addresses this issue. Please check the latest commit.

File: src/plugins/home/public/application/components/use_case_card.ts (hunk 7 above)
Comment (Qxisylolo) [reply src/plugins/home/public/application/components/use_case_card.ts:30]: Hi，2 overview page for observability and security-analysis haven't been finished, but I will make it clickable and set the navigation. Thanks

File: src/plugins/home/public/application/home_render.tsx [hunk 8]
```diff
@@ -47,28 +45,52 @@ export const setupHome = (contentManagement: ContentManagementPluginSetup) => {
...
+  useCases.forEach((useCase, index) => {
+    registerUseCaseCard(contentManagement, workspaceEnabled, {
+      id: useCase.id,
+      order: index + 1,
+      description: useCase.description,
+      title: useCase.title,
+      target: HOME_CONTENT_AREAS.GET_STARTED,
+      icon: useCase.icon ?? '',
+    });
+  });
```
Comment (ruanyl) [inline src/plugins/home/public/application/home_render.tsx:81]:
```suggestion
if (workspaceEnabled) {
  const useCases = [
    DEFAULT_NAV_GROUPS.observability,
    DEFAULT_NAV_GROUPS.search,
    DEFAULT_NAV_GROUPS['security-analytics'],
  ];

  useCases.forEach((useCase, index) => {
    registerUseCaseCard(contentManagement, {
      id: useCase.id,
      order: index + 1,
      description: useCase.description,
      title: useCase.title,
      target: HOME_CONTENT_AREAS.GET_STARTED,
      icon: useCase.icon ?? '',
    });
  });
}
```

File: src/plugins/home/public/application/components/use_case_card.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  }: {
+    target: string;
+    order: number;
+    id: string;
+    title: string;
+    description: string;
+    icon: string;
+  }
+) => {
+  if (workspaceEnabled) return;
```
Comment (ruanyl) [inline src/plugins/home/public/application/components/use_case_card.ts:29]: Move `workspaceEnabled` check out to where `registerUseCaseCard` was called and do a if condition from there.

Comment (Qxisylolo) [reply src/plugins/home/public/application/components/use_case_card.ts]: Thanks for the comments, already updated.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Comment by SuZhou-Joe** [comment]
`Run cypress tests (osd:ciGroup6)` should be irrelevant to the code changes.

**Pull request merged**